frontend/src/containers/DocList.js

```diff
@@ -13,45 +14,64 @@ import Container from '../components/Container';
 import DocumentList from '../components/app/DocumentList';
 import Overlay from '../components/app/Overlay';
 
+const EMPTY_ARRAY = [];
+const EMPTY_OBJECT = {};
+
 /**
  * @file Class based component.
  * @module DocList
  * @extends Component
  */
 class DocList extends Component {
+  shouldComponentUpdate(nextProps) {
+    const { query } = this.props;
+    const { query: nextQuery } = nextProps;
+
+    if (_.isEqual(query, nextQuery)) {
```

Comment:
All this could be written as `return _.isEqual(query, nextQuery) ? false : true;`

Reply:
or...

`return ! _.isEqual(query, nextQuery);`